cmd/run.go

```diff
@@ -34,7 +34,16 @@ log and return exit code of the command.`,
 		}
 
 		if err := e.Run(tableName, keyName, shellCommand, sleepStartRandom, holdLockBy); err != nil {
-			logrus.Fatal(err)
+			// dynamolock.LockNotGrantedError doesn't export the error so this is the next best we can do.
+			// This is obviously super fragile (i.e if the error msg wording is updated, this fails)
+			lockFailedErr := "Didn't acquire lock because it is locked and request is configured not to retry."
```

Comment:
thoughts on moving this logic within within `exec.Run` and returning `err` as `nil` and logging as a warning from there itself?

Reply:
Good idea, that feels cleaner.